src/sparseml/modifiers/obcq/base.py

```diff
@@ -61,3 +61,27 @@ class SparseGPTModifier(Modifier):
 
     def on_initialize_structure(self, state: "State", **kwargs):
         pass  # nothing needed for this modifier
+
+    def _validate_layerwise_sparisity(self):
+        if isinstance(self.sparsity, float):
+            return  # single sparsity will be applied to all layers
+
+        if not isinstance(self.targets, List):
+            raise ValueError(
+                "Layer targets must be a list when specifying layer-wise"
+                f" sparsity. Got {self.targets}"
+            )
+
+        if len(self.targets) != len(self.sparsity):
+            raise ValueError(
+                "Number of layer targets must match the number of "
+                f"sparsities. Got {len(self.targets)} layers and "
+                f"{len(self.sparsity)} sparsities"
+            )
+
+        for layer_name in self.targets:
+            if layer_name.startswith("re:"):
```

Comment:
why can't we support this? for overlaps?

Reply:
Yes, this is to avoid a situation where a layer matches multiples regex's and gets assigned multiple sparsities. We could definitely handle this case with some work but it didn't seem worth the extra complication